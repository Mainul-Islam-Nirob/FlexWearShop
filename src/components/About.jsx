import React from 'react';

function About() {
  return (
    <div className="min-h-screen bg-gray-50 py-16 px-4">
      <div className="max-w-4xl mx-auto bg-white p-10 rounded-xl shadow-md">
        <h2 className="text-4xl font-bold text-center text-gray-800 mb-6">About FlexWearShop</h2>

        <p className="text-gray-600 text-lg leading-relaxed mb-4">
          Welcome to <span className="font-semibold text-blue-600">FlexWearShop</span> — your one-stop destination for fashion that’s flexible, fresh, and fabulous.
        </p>

        <p className="text-gray-600 text-lg leading-relaxed mb-4">
          We’re passionate about helping you express your personal style with quality clothing, shoes, and accessories. Whether you're dressing for comfort, performance, or making a statement, we have something for everyone.
        </p>

        <p className="text-gray-600 text-lg leading-relaxed mb-4">
          Our mission is simple: to deliver premium fashion with a seamless shopping experience. From handpicked collections to smooth checkout, we make fashion fun and easy.
        </p>

        <p className="text-gray-600 text-lg leading-relaxed">
          Thank you for choosing <span className="font-semibold text-blue-600">FlexWearShop</span>. We’re here to help you look great and feel confident—every single day.
        </p>
      </div>
    </div>
  );
}

export default About;
