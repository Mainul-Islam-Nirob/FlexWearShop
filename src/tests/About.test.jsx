import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import About from '../components/About'; // adjust path if needed

describe('About component', () => {
  it('renders without crashing', () => {
    render(<About />);
    const heading = screen.getByRole('heading', { name: /about flexwearshop/i });
    expect(heading).toBeInTheDocument();
  });

  it('displays all expected paragraphs', () => {
    render(<About />);
    
    expect(
      screen.getByText(/your one-stop destination for fashion/i)
    ).toBeInTheDocument();
    
    expect(
      screen.getByText(/express your personal style/i)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/premium fashion with a seamless shopping experience/i)
    ).toBeInTheDocument();

    expect(
      screen.getByText(/look great and feel confident/i)
    ).toBeInTheDocument();
  });

  it('highlights "FlexWearShop" brand name in multiple places', () => {
    render(<About />);
    const brandMentions = screen.getAllByText(/FlexWearShop/i);
    expect(brandMentions.length).toBeGreaterThanOrEqual(2);
  });

  it('matches snapshot', () => {
    const { container } = render(<About />);
    expect(container).toMatchSnapshot(); // optional â€” useful if UI is stable
  });
});
