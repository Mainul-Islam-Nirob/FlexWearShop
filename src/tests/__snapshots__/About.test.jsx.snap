// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About component > matches snapshot 1`] = `
<div>
  <div
    class="min-h-screen bg-gray-50 py-16 px-4"
  >
    <div
      class="max-w-4xl mx-auto bg-white p-10 rounded-xl shadow-md"
    >
      <h2
        class="text-4xl font-bold text-center text-gray-800 mb-6"
      >
        About FlexWearShop
      </h2>
      <p
        class="text-gray-600 text-lg leading-relaxed mb-4"
      >
        Welcome to 
        <span
          class="font-semibold text-blue-600"
        >
          FlexWearShop
        </span>
         — your one-stop destination for fashion that’s flexible, fresh, and fabulous.
      </p>
      <p
        class="text-gray-600 text-lg leading-relaxed mb-4"
      >
        We’re passionate about helping you express your personal style with quality clothing, shoes, and accessories. Whether you're dressing for comfort, performance, or making a statement, we have something for everyone.
      </p>
      <p
        class="text-gray-600 text-lg leading-relaxed mb-4"
      >
        Our mission is simple: to deliver premium fashion with a seamless shopping experience. From handpicked collections to smooth checkout, we make fashion fun and easy.
      </p>
      <p
        class="text-gray-600 text-lg leading-relaxed"
      >
        Thank you for choosing 
        <span
          class="font-semibold text-blue-600"
        >
          FlexWearShop
        </span>
        . We’re here to help you look great and feel confident—every single day.
      </p>
    </div>
  </div>
</div>
`;
